### PR TITLE
[SMAGENT-4569] New PodDisruptionBudget KSM metrics

### PR DESCRIPTION
**Pull request by bawhetst**

### CODE REVIEW

**Review by balinaveen — COMMENTED** [review]
@bawhetst  Please note that these enhancements need to be ported to our helm charts too. cc @claffin @aroberts87

**Comment by bawhetst** [comment]
I've opened the Helm PR at https://github.com/sysdiglabs/charts/pull/968.

**Review by nathan-b — APPROVED** [review]
LGTM

File: agent_deploy/kubernetes/sysdig-agent-clusterrole.yaml [hunk 1]
```diff
@@ -96,6 +96,14 @@ rules:
   - get
   - list
   - watch
+- apiGroups:
+  - policy
+  resources:
+  - poddisruptionbudgets
```
Comment (balinaveen) [inline agent_deploy/kubernetes/sysdig-agent-clusterrole.yaml:102]: How much cpu/mem impact does this have on the agent? I assume these don't change often, so possibly mostly  a small memory cost. Is that right?

**Review by mbreitung — APPROVED** [review]

**Review by balinaveen — APPROVED** [review]
LGTM

**Review by mstemm — APPROVED** [review]
Looks fine, I guess. The more important change is good reporting of an error in the agent if it can't fetch these resources, though. I thought it was very rare to change the clusterrole for the agent once installed, so any existing customers won't be updating their permissions for their agents but will have agents trying to fetch these resources right?

Maybe with helm it always reapplies the cluster role, I don't know.

**Pull request merged**